katsdpingest/ingest_session.py

```diff
@@ -165,6 +165,23 @@ def _split_array(x: np.ndarray, dtype) -> np.ndarray:
     return np.asarray(np.lib.stride_tricks.DummyArray(interface, base=x))
 
 
+def _fast_unique(x: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
+    """Faster version of :func:`np.unique` for specific case.
+
+    It is equivalent to ``np.unique(x, axis=0, return_inverse=True)``, where
+    ``x`` must be a 2D array.
+    """
+    # Implementation is based on
+    # https://github.com/numpy/numpy/issues/11136#issue-325345618. In short, each
+    # row is viewed as a byte string, which for some reason makes it hundreds of
+    # times faster.
+    x = np.ascontiguousarray(x)
+    new_dtype = np.dtype(f'S{x.dtype.itemsize * x.shape[1]}')
```

Comment:
`new_dype` has a specifier for zero-terminated bytes `(f'S{%}')` rather than unicode strings: do we anticipate Python3 compatibility issues?

Reply:
It's not actually text - it's just making a dtype of the appropriate size to view a chunk of binary data.

But when you say "zero-terminated" it's got me thinking that maybe it won't be guaranteed to work properly when there are zeroes in the data (although the current test cases should pick that up). The bug I referenced has an alternative using `np.void`. Let me take a look at that and see if the performance is the same.

Reply:
It seems to be even faster with `np.void`, so I've updated to that.